app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModel.kt

```diff
@@ -129,8 +129,13 @@ class ZimManageViewModel @Inject constructor(
   }
 
   override fun onCleared() {
-    compositeDisposable.clear()
+    compositeDisposable?.clear()
     context.unregisterReceiver(connectivityBroadcastReceiver)
+    connectivityBroadcastReceiver.stopNetworkState()
+    requestFileSystemCheck.onComplete()
+    fileSelectActions.onComplete()
+    requestDownloadLibrary.onComplete()
+    compositeDisposable = null
```

Comment:
> Even though we've added what looks like proper disposal in onCleared(), our retained Disposable instance is, itself, a leak because it's still transitively holding onto a reference to anything captured in the lambda!

The reason why I set `CompositeDisposable` to nullable. More info available on  [this](https://www.zacsweers.dev/disposables-can-cause-memory-leaks) article.